java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -81,36 +81,29 @@ public Pop(ProgramState programState, List<SymbolicValue> result) {
           .put(BooleanConstraint.class, BooleanConstraint.FALSE).put(ObjectConstraint.class, ObjectConstraint.NOT_NULL)),
     PCollections.emptyMap(),
     PCollections.emptyStack(),
-    null);
+    null, PCollections.emptyMap());
 
   private final PMap<ProgramPoint, Integer> visitedPoints;
-
-  @Nullable
-  Symbol lastEvaluated;
-
   private final PStack<SymbolicValue> stack;
   private final PMap<SymbolicValue, Integer> references;
   private SymbolicValue exitSymbolicValue;
   final PMap<Symbol, SymbolicValue> values;
   final PMap<SymbolicValue, PMap<Class<? extends Constraint>, Constraint>> constraints;
+  final PMap<SymbolicValue, Symbol> lastAssociatedSymbol;
```

Comment:
I think we should add the plural to this variable name: `lastAssociatedSymbols`.